onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -303,8 +303,12 @@ common::Status NnapiExecutionProvider::Compile(const std::vector<onnxruntime::No
         OperandType input_type = model_input_type;
         input_type.SetDimensions(dimensions);
 
-        if (input_type.GetOperandBlobByteSize() == 0)
-          return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "The actual input cannot have 0 dim (dynamic)");
+        // We have some op has input can have {0} shapes, such as Resize.scales/roi,
+        // before these input be turned into optional input, these inputs are valid
+        if (input_type.GetOperandBlobByteSize() == 0) {
+          LOGS_DEFAULT(WARNING) << "The actual input [" << input_name << "] has "
+                                << nnapi::Shape2String(dimensions) << " shape";
+        }
 
```

Comment:
Is it possible to track the names of inputs that are allowed to have empty data? This warning message would be somewhat obtuse to interpret if seen in a log as it's not clear if/when there's an actual problem.

Reply:
Changed to since this is a valid input now, (it was not before we add resize)

---
In reply to: [502982296](https://github.com/microsoft/onnxruntime/pull/5427#discussion_r502982296) [](ancestors = 502982296)